01week/datatypes.js

```diff
@@ -0,0 +1,186 @@
+    return parseInt(arg1, 10);
+  } else {
+    return "Argument provided is not a String type";
+  }
+};
+
+console.log(convertStrToNum("93"));
+// check typeof value returned by convertStrToNum
+console.log(typeof convertStrToNum("93"));
+// try passing an argument that is NOT a string type
+console.log(convertStrToNum(93));
+
+console.log(); // blank line separator
+
+// 4. write a js function that takes in different datatypes and prints out the datatype of the arg
+//
+// I was going to make this a oneline function that returns the typeof the arg passed to it
+// however, typeof null = object and typeof NaN = number
+// so I will expand the function to be more explicit in it's output
+//
+// first check the data types that are truthy
+// second check the data types that are falsy
+//  - this is bc some truthy types when checked will return falsy in functions like isNaN
+//  - ex: isNaN("this is a string") will return true, so should be placed AFTER the string check
+
+const checkDataType=(arg1)=>{
+  // begin checking TRUTHY data types
+  // check for Boolean type
+  if (typeof arg1 === "boolean") {
+    console.log(arg1 + " is type Boolean");
+  }
+  // check for Number type
+  // we need to check both typeof arg1 && isNaN(arg1) === "false" bc typeof NaN === "number"
+  else if (typeof arg1 === "number" && isNaN(arg1) === false) {
+    console.log(arg1 + " is type Number");
+  }
+  // check for String type
+  else if (typeof arg1 === "string") {
+    console.log(arg1 + " is type String");
+  }
+  // begin checking FALSY data types
+  // check for null type
+  else if (arg1 === null) {
+    console.log(arg1 + " is type Null");
+  }
+  // check for Undefinded type
+  else if (arg1 === undefined) {
+    console.log(arg1 + " is type Undefined");
+  }
+  // check for NaN type using isNaN method
+  else if (isNaN(arg1)) {
+    console.log(arg1 + " is type NaN");
+  }
+};
+
+// check Boolean type
+checkDataType(true);
+// check Null type
+checkDataType(null);
+// check Undefinded type
+checkDataType(undefined);
+// check Number type
+checkDataType(31);
+// check NaN type
+// parseInt("string not a number") will return a value of NaN
+checkDataType(parseInt("string not a number"));
+// check String type
+checkDataType("Howdydoodee");
+
+console.log(); // blank line separator
+
+// 5. write a js function that adds two numbers together
+//
+// oneline function that returns the addition of two arguments
+
+const addTwoNumbers=(arg1, arg2)=> arg1 + arg2;
+
+console.log(addTwoNumbers(20, 40));
+
+console.log(); // blank line separator
+
+// 6. write a js function that runs only when 2 'things' (args) are true
+//
+// the wording of the problem says runs only when 2 things are "true" and not "truthy"
+// so I assume the expected arguments are boolean type
+//
+// check both args given to see if they === true, if so, print message
+// if 1 or more args !== true, do nothing
+
+const twoArgsTrue=(arg1, arg2)=>{
+  if (arg1 === true && arg2 === true) {
```

Comment:
This evaluates if the arguments are the boolean `true`. I need a function that evaluates if both inputs evaluate to true.